### DISCUSSION TITLE
Can I set externalURL in the argo Application yaml?

### DISCUSSION THREAD

**Discussion by SamDecrock** [argoproj/argo-cd#5230] [Q&A]
I see that externalURLs is automatically generated based on the ingress rules. However, my application has several ingress rules and doesn't pick the main one.

Can I somehow specify the externalURL in the manifest of the ArgoCd Application?

**Comment by sbose78** [comment]
@SamDecrock The current Application CRD doesn't have a field to do so. 

>  However, my application has several ingress rules and doesn't pick the main one.

Do you imply that your Application consists of an Ingress resource with multiple rules ?

**Reply by SamDecrock** [reply]
yes, my application has 4-5 ingress rules, each for different parts of the application.

**Comment by shake82** [comment]
In the same situation. I am using Istio Gateways for Ingress and the Urls are not picked up. Would be nice to be able to set them through configuration if possible

**Comment by kshamajain99** [comment]
You can add external-link annotation to add externalURL
Example: #4380

**Reply by SamDecrock** [reply]
thanks. In what version did this end up?

Ah, https://github.com/argoproj/argo-cd/releases/tag/v1.8.0 
never mind, thx!

**Comment by SamDecrock** [comment]
Actually I'm looking for a way to override the ones extracted from the ingress rules.

**Reply by nufissime** [reply]
Hey, I know it's been some time but did you manage to find a way to do it ?

Sure, we can use "link.argocd.argoproj.io/external-link: http://my-grafana.com/pre-generated-link" to add another URL, but I have useless ingress rules that I'd like to get rid off in the UI.
